contributing_to_docs/doc_guidelines.adoc

```diff
@@ -217,8 +217,82 @@ Considering that we use distinct branches to keep content for product versions s
+== Node name formatting
 
+Do not use internal company server names in commands or example output. Provide generic OpenShift Container Platform node name examples that are not provider-specific, unless required. Where possible, use the example.com domain name when providing fully qualified domain names (FQDNs).
+
+The following table includes example OpenShift Container Platform 4 node names and their corresponding role types:
+
+[options="header"]
+|===
+
+|Node name |Role type
+
+|*node-1.example.com*
+.3+.^|You can use this format for nodes that do not need role-specific node names.
+
+|*node-2.example.com*
+
+|*node-3.example.com*
+
+|*control-plane-1.example.com*
+.3+.^|You can use this format if you need to describe the control plane role type within a node name.
+
+|*control-plane-2.example.com*
+
+|*control-plane-3.example.com*
+
+|*compute-1.example.com*
+.2+.^|You can use this format if you need to describe the compute node role type within a node name.
+
+|*compute-2.example.com*
+
+|*bootstrap.example.com*
+|You can use this format if you need to describe the bootstrap node role type within a node name.
+|===
+
+This example lists the status of cluster nodes that use the node name formatting guidelines:
+
+....
+[source,terminal]
+----
+$ oc get nodes
+----
++
+.Example output
+[source,terminal]
+----
+NAME                          STATUS   ROLES    AGE   VERSION
+compute-1.example.com         Ready    worker   33m   v1.19.0+9f84db3
+control-plane-1.example.com   Ready    master   41m   v1.19.0+9f84db3
+control-plane-2.example.com   Ready    master   45m   v1.19.0+9f84db3
+compute-2.example.com         Ready    worker   38m   v1.19.0+9f84db3
+compute-3.example.com         Ready    worker   33m   v1.19.0+9f84db3
+control-plane-3.example.com   Ready    master   41m   v1.19.0+9f84db3
+----
+....
+
+[NOTE]
+====
+Some provider-formatted host names include IPv4 addresses. An OpenShift Container Platform node name typically reflects the host name of a node. If node names in your output need to be provider-specific and require this format, use private IPv4 addresses. For example, you could use `ip-10-0-48-9.example.com` as a node name that includes a private IPv4 address.
+====
+
+== IPv4 address formatting
+
+Do not use public IPv4 addresses in commands or example output in the product documentation. If you include the output from a command run on an OpenShift Container Platform test cluster in the documentation, replace any public or sensitive IPv4 addresses. You can replace public IPv4 addresses with addresses in the 203.0.113.0/24 TEST-NET-3 address block, which range from 203.0.113.0 to 203.0.113.255. The TEST-NET-3 address block is reserved for documentation. See link:https://tools.ietf.org/html/rfc5737[IETF RFC 5737 - IPv4 Address Blocks Reserved for Documentation] for more information about IPv4 address ranges that are reserved for documentation.
```

Comment:
Why aren't we using the IP address range that the style guide requires?

> * 192.0.2.0/24 (or 192.0.2.*)
> * 198.51.100.0/24 (or 198.51.100.*)
> * 203.0.113.0/24 (or 203.0.113.*)

Reply:
+1. Not sure why we're deviating from the ranges set in the style guide.

Reply:
I might be missing something here, but it looks like `203.0.113.0/24` **is** the range mentioned in the style guide (3rd bullet). The rest of the sentence (`which range from 203.0.113.0 to 203.0.113.255`) is an explanation. The explanation might be overkill, however, because I would expect a contributor to be familiar with CIDR notation.

Reply:
Thank you for your input!

During a discussion last week, @mikemckiernan highlighted that the 192.0.2.0/24 TEST-NET-1 and 198.51.100.0/24 TEST-NET-2 documentation addresses might look too similar to the 192.168.0.0/16 private address range. Since we are suggesting reserved documentation addresses as a replacement for public IPs, we put forward a recommendation to use the 203.0.113.0/24 TEST-NET-3 documentation address block specifically. That block does not look similar to the private address range.

Also, this PR initially suggested using the TEST-NET addresses for example output. @mikemckiernan highlighted that it would be better to use the private IP addresses that are output by OpenShift by default.

So, the current proposal essentially highlights that it is ok to use the private IP addresses included in output from test clusters, but provides guidance on how to replace public IPs, since they might resolve to a real world node.

Please let me know if you would like me to add further clarification within the content about this.

Reply:
The private ranges are larger than the default cluster network and default service network, 10.128.0.0/14 and 172.30.0.0/16, respectively. The clusters that associates access use the default values and my feeling is that we want to repeat them in product documentation to reinforce customer understanding.

The use of the 203.0.113.0/24 range to replace public IP addresses provides good visual distinction from the private ranges.  My expectation is customers can quickly scan sample output, see a 203 range IP address, and recognize that their output should differ--according to the cloud provider their cluster uses.

Reply:
I haven't seen any indications that this is taxing on customers, but it would generate some additional work for us. 

I'd prefer to keep the ISG convention. The change seems like it could increase cognitive load for writers to address something that, so far, hasn't popped up as a customer concern. 

Is the proposal that this rule would be a "going forward" thing, or would every actively developed version of the docs be updated upon its implementation?

Reply:
@maxwelldb thank you for your comment. Can you please let me know which part of the ISG convention you are referring to? I want to ensure that I understand your preference here properly. Thanks!

Reply:
@pneedle-rh The IPv4 section in "Chapter 5 - References" -> "References to web addresses, protocols, and IP addresses."

Reply:
@maxwelldb so is my understanding correct then that your preference is for the following, from the **IPv4 addresses** subsection in that chapter?:

"If your example already contains the loopback address and you need additional example
IPv4 addresses, use addresses in the /8 IP address block that are reserved for Test-Net: 192.0.2.0
through 192.0.2.24."

Reply:
@pneedle-rh, the new version of the style guide does not mention the Test-Net ranges.

~~~
You can also use the IP loopback address, which is 127.0.0.0/8 (127.*.*.*), although typically, only 127.0.0.1 is used for loopback purposes.
You can include wildcard characters in examples of IPv4 addresses to illustrate that a range of values is permitted; for example, 127.*.*.12 includes all addresses that start with 127 and end with 12.
If you use wildcards, do not use identifiable companies' addresses in documentation. For example, using 192.*.*.12 is permissible because the 192.0.0.0/8 range covers a variety of uses.
~~~

Reply:
@pneedle-rh Oh, I was using the paper guide. In the newest version (p. 258 of the PDF that's in circulation): 

> If you must use an IP address in an example, use an address that is reserved for examples, and not one that relates to a particular company, unless that is required. For example, all addresses in the range 9.0.0.0/8 (or 9.\*.\*.\*) are owned by IBM. Three ranges of addresses are reserved for use in documentation:
> 
> * 192.0.2.0/24 (or 192.0.2.*)
> * 198.51.100.0/24 (or 198.51.100.*)
> * 203.0.113.0/24 (or 203.0.113.*)
> 
> ... 
> 
> You can also use the IP loopback address, which is 127.0.0.0/8 (127.\*.\*.\*), although typically, only 127.0.0.1 is used for loopback purposes. 
> 
> You can include wildcare characters in examples of IPv4 addresses to illustrate that a range of values is permittedl; for example, 127.\*.\*.12 includes all addresses that start with 127 and end with 12. If you use wildcards, do not use identifiable companies' addresses in documentation. For example, using 192.\*.\*.12 is permissible because the 192.0.0.0/8 range covers a variety of uses.

Reply:
@kalexand-rh and @maxwelldb thank you. It looks like the version of the ISG I have is out of date. I will update my copy.

The 192.0.2.0/24, 198.51.100.0/24 and 203.0.113.0/24 that are referenced in Max's updated ISG example are the three TEST-NET ranges, as detailed in https://tools.ietf.org/html/rfc5737.

Thank you both for clarifying your preference for the guidelines.